pipelines/shopgate.user.addFavoriteData.v1.json

```diff
@@ -0,0 +1,155 @@
+    "public": true,
+    "input": [
+      {
+        "key": "productIds",
+        "id": "1"
+      },
+      {
+        "key": "products",
+        "id": "100"
+      }
+    ],
+    "output": [
+      {
+        "key": "magentoParentProducts",
+        "id": "150",
+        "optional": true
+      },
+      {
+        "key": "products",
+        "id": "100"
+      }
+    ],
+    "steps": [
+      {
+        "type": "pipeline",
+        "id": "shopgate.general.getToken.v1",
+        "trusted": true,
+        "input": [],
+        "output": [
+          {
+            "key": "token",
+            "id": "140"
+          }
+        ]
+      },
+      {
+        "type": "extension",
+        "id": "@shopgate/magento-cart",
+        "path": "@shopgate/magento-cart/products/requestParentProductsFromMagento.js",
+        "input": [
+          {
+            "key": "productIds",
+            "id": "1"
+          },
+          {
+            "key": "products",
+            "id": "100"
+          },
+          {
+            "key": "token",
+            "id": "140"
+          }
+        ],
+        "output": [
+          {
+            "key": "magentoParentProducts",
+            "id": "150",
+            "optional": true
+          }
+        ]
+      },
+      {
+        "type": "extension",
+        "id": "@shopgate/magento-cart",
+        "path": "@shopgate/magento-cart/products/appendProductVariantsMetadataToProductList.js",
+        "input": [
+          {
+            "key": "products",
+            "id": "100"
+          },
+          {
+            "key": "magentoParentProducts",
+            "id": "150",
+            "optional": true
+          }
+        ],
+        "output": [
+          {
+            "key": "products",
+            "id": "100"
+          }
+        ]
+      },
+      {
+        "type": "extension",
+        "id": "@shopgate/magento-favorites",
+        "path": "@shopgate/magento-favorites/lib/transformItemId.js",
+        "input": [
+          {
+            "key": "products",
+            "id": "100"
+          }
+        ],
+        "output": [
+          {
+            "key": "products",
+            "id": "100"
+          }
+        ]
+      },
+      {
+        "type": "extension",
+        "id": "@shopgate/magento-cart",
+        "path": "@shopgate/magento-cart/cart/createProductCartItemList.js",
+        "input": [
+          {"key": "products", "id": "100"}
+        ],
+        "output": [
+          {"key": "transformedProducts", "id": "150"}
+        ]
+      },
+      {
+        "type": "extension",
+        "id": "@shopgate/magento-favorites",
+        "path": "@shopgate/magento-favorites/lib/magento/customer/getWishlistId.js",
+        "input": [
+          {
+            "key": "token",
+            "id": "140"
+          }
+        ],
+        "output": [
+          {
+            "key": "wishlistId",
+            "id": "110"
+          }
+        ]
+      },
+      {
+        "type": "extension",
```

Comment:
I like what you had done. I think this addItems step needs to be moved out. You can move it out to a pipeline called `addUserFavoriteItems.v1` which calls this current pipeline and adds an item. 
Why this is needed is to also use this current pipeline implementation for the existing getFavorites pipeline. Essentially, we would like to stop using the BigAPI to retrieve products for the favorites extension.

Reply:
due to time constraints, this may need to be moved to the next story

Reply:
Ok, we should do that when we implement the pipeline for put Favorites. I guess we planned enough time for that because we can use functionality from this implementation.

Reply:
Ok, we should do that when we implement the pipeline for put Favorites. I guess we planned enough time for that because we can use functionality from this implementation.

Reply:
Ok, we should do that when we implement the pipeline for put Favorites. I guess we planned enough time for that because we can use functionality from this implementation.